preppal-fe/src/__tests__/pages/recipes.test.tsx

```diff
@@ -2,65 +2,26 @@ import { render, screen } from "@testing-library/react";
-    ingredients: ["ingredient 1", "ingredient 2"],
-    instructions: ["step 1", "step 2"],
-    servingSize: 1,
-    prepTime: 20,
-    cookingTime: 30,
-    creationDate: new Date("2024-02-17T21:49:08.000Z")
-}
-
-const mockUseLocation = {
-    pathname: '/testApiSandboxAccount/test-recipe-1000',
-    key: '',
-    search: '',
-    hash: '',
-    state: { recipeInfo },
-};
-
-jest.mock("react-router-dom", () => ({
-    ...jest.requireActual("react-router-dom"),
-    useLocation: () => mockUseLocation,
-}));
-
-describe('Collections page tests', () => {
-    test('Render standard component --> headers and data', async () => {
+describe('Recipe tests', () => {
+    test('Render standard component --> headers', async () => {
         render(<BrowserRouter><Recipe /></BrowserRouter>);
-
-        const title = screen.queryByText("Chicken's Fried Rice");
-        const desc = screen.queryByText("A chicken fried this rice.");
-        const author = screen.queryByText("Author: John Doe");
-        const date = screen.queryByText("Date published: Feb 17 2024");
-        const ingredientsHeader = screen.queryByText("Ingredients");
-        const ingredient1 = screen.queryByText("ingredient 1");
-        const ingredient2 = screen.queryByText("ingredient 2");
-        const instructionsHeader = screen.queryByText("Instructions");
-        const step1 = screen.queryByText("step 1");
-        const step2 = screen.queryByText("step 2");
-        const servings = screen.queryByText("Serving Size: 1");
-        const prepTime = screen.queryByText("Prep Time: 20 min");
-        const cookingTime = screen.queryByText("Cooking Time: 30 min");
-        const totalTime = screen.queryByText("Total Time: 50 min");
-
-        expect(title).toBeTruthy();
-        expect(desc).toBeTruthy();
+        const author = screen.queryByText("Author:");
         expect(author).toBeTruthy();
-        expect(date).toBeTruthy();
-        expect(ingredientsHeader).toBeTruthy();
-        expect(ingredient1).toBeTruthy();
-        expect(ingredient2).toBeTruthy();
-        expect(instructionsHeader).toBeTruthy();
-        expect(step1).toBeTruthy();
-        expect(step2).toBeTruthy();
+        const ingredients = screen.queryByText("Ingredients");
+        expect(ingredients).toBeTruthy();
+        const instructions = screen.queryByText("Instructions");
+        expect(instructions).toBeTruthy();
+        const servings = screen.queryByText("Serving Size:");
         expect(servings).toBeTruthy();
-        expect(prepTime).toBeTruthy();
-        expect(cookingTime).toBeTruthy();
-        expect(totalTime).toBeTruthy();
+    });
+
+    test('Render standard component --> reviews', async () => {
+        render(<BrowserRouter><Recipe /></BrowserRouter>);
+
+        const leaveReview = screen.queryByText("Leave a review");
+        expect(leaveReview ).toBeTruthy();
+        const reviews = screen.queryByText("Reviews");
+        expect(reviews).toBeTruthy();
     });
```

Comment:
should buttons be tested for? could just check the text of them maybe and if they're disabled

Reply:
The entire leave a review section is hidden if the user is not logged in so it might might be possible to look for a submit button.